api/README.md

```diff
@@ -28,6 +28,9 @@ Create a table `containers` with following options:
 
 # Restrict memory usage of containers
 memory_limit = '1g'
+# Optional field to Enable/Disable automatic docker volume mounting for containers. 
+# Defaults to false if no option provided. (Accepted values - true, false)
+docker_volume_enabled = false
```

Comment:
First of all, thanks for adding such an option to be backward compatible. But I have some obligations with this option:

- The name suggests it works only in Docker environment but you are working on #123 for all backends. So this name is misleading, IMHO.
- It is a boolean and therefore not very descriptive
- It is a global option and there is no control to enable it for specific services. I would like to see that at companion level.

What do you think of following config sketch?

```toml
[companions.postgres]
type = 'application'
image = 'postgres'

[companions.postgres.volumes]
# Alternative value: MountDeclaredImageVolumes
volumeMounts = 'none'
```

Later `volumeMounts` could be extended so that it accepts a list of paths so that the user has control over which volumes will be mounted. This approach might help in the planning extension of hooks that manipulate volume configurations.

Reply:
After looking at the code I realized this would cause a conflict. Currently `.volumes` alias is occupied by the `files` variable, meaning straight forward adding under volumes means that `volumeMounts` should also apply to `files`.
I don't think that is what you are looking for.

- [ ] I can add some complexity to handle `volumeMounts` only if it is under `.volumes` and is ignored if under `.files` which according to me gives a better meaning.

- [ ] Alternative approach is to maintain `volumeMounts` as a new variable under companions. Meaning it would be referred to as `companions.postgres.volumeMounts` without any dependency on `volumes` or `files`.

Let me know which approach makes better sense.